docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,124 @@
+# API 20201231-preview design
+
+For api `20201231-preview` bellow is the list of proposed changes and features.
+These would have to land into API and backend as a features.
+
+Feature work will need to be agreed and converted to working stories. All technical details bellow are overview only
+
+## API changes
+
+### .OpenShiftClusterProperties.SecurityProfile
+
+Add security profile for all security enhancements for the product.
+Proposal is to use new profile structure, so we do not scatter these options
+to corresponding objects. In example, `EncryptionAtHost` can be set on individual
+worker pools. But this creates possibility to configure some nodes with/without
+encryption.
+
+We need to make sure that these option are validated on the cluster too when customer is interacting with MachineSet objects. We should be able to verify
+if these options where enabled on cluster create and set those accordingly.
+
+This validation should be part of machine-api webhooks.
```

Comment:
I don't think this is relevant to this document.  Separately, I'm somewhat sceptical about machine-api webhooks because the RP/installer generates all the machinesets before there's a webhook to call.  I do think it makes sense for the RP to validate the basics up front, however.